medicines/doc-index-updater/src/service_bus_client/mod.rs

```diff
@@ -70,8 +70,31 @@ pub struct RetrievedMessage<T: Message> {
     peek_lock: PeekLockResponse,
 }
 
-impl<T: Message> RetrievedMessage<T> {
-    pub async fn remove(&self) -> Result<String, anyhow::Error> {
+#[async_trait]
+pub trait Removeable {
+    async fn remove(&mut self) -> Result<String, anyhow::Error>;
+}
+
+#[cfg(test)]
+pub struct ShouldRemove {
```

Comment:
rename this to `TestRemoveable` - it shouldn't have opinions about whether it should or should not be removed